### PR TITLE
fix: allow local FunASR without API key

### PR DESCRIPTION
**Pull request by LauraGPT**
### Summary

The FunASR provider added in #17171 defaults to a local `http://localhost:8000/v1` server, but it still inherited the global API-key requirement and unconditionally built an Authorization header. This prevented the default unauthenticated self-hosted deployment from working. The transcription path also dereferenced a missing model name while building its multipart request.

This change:

- allows an empty API key for FunASR, matching other local providers
- omits the Authorization header when no key is configured while preserving trimmed Bearer authentication when one is provided
- validates and trims the ASR model name before building the multipart request, returning an error instead of panicking
- adds HTTP-level regression coverage for unauthenticated transcription/model listing and optional authentication

### User impact

Users can connect RAGFlow to the default local FunASR server without inventing a placeholder API key. Authenticated compatible endpoints continue to receive a standard Bearer header.

### Tests

- `go test ./internal/entity/models -run '^TestFunASR' -count=1 -v`
- `go test ./internal/entity/models -count=1`
- `git diff --check`

### CODE REVIEW

File: internal/entity/models/funasr_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			t.Errorf("method=%s, want POST", r.Method)
+		}
+		if r.URL.Path != "/audio/transcriptions" {
+			t.Errorf("path=%s, want /audio/transcriptions", r.URL.Path)
+		}
+		if got := r.Header.Get("Authorization"); got != "" {
+			t.Errorf("Authorization=%q, want no header", got)
+		}
+		if err := r.ParseMultipartForm(1 << 20); err != nil {
+			t.Fatalf("parse multipart form: %v", err)
```
[the remark this answers is not in the document]
Comment (LauraGPT) [reply internal/entity/models/funasr_test.go]: Fixed in b70896153: the handler records the multipart parse failure with t.Errorf and returns immediately. The complete internal/entity/models package passes.

File: internal/entity/models/funasr_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			t.Errorf("model=%q, want fun-asr-nano", got)
+		}
+		_ = json.NewEncoder(w).Encode(map[string]string{"text": "hello"})
+	}))
+	defer srv.Close()
+
+	modelName := "fun-asr-nano"
+	file := writeFunASRTestAudio(t)
+	resp, err := newFunASRForTest(srv.URL).TranscribeAudio(
+		context.Background(), &modelName, &file, &APIConfig{}, nil, nil,
```
[the remark this answers is not in the document]
Comment (LauraGPT) [reply internal/entity/models/funasr_test.go]: Fixed in b70896153: the success case now verifies surrounding whitespace is serialized as fun-asr-nano, and the missing-name test covers both nil and whitespace-only values. The complete internal/entity/models package passes.

**Comment by LauraGPT** [comment]
@JinHai-CN, could you review this small follow-up to the FunASR provider from #17171 when convenient?

It fixes the default local deployment path: an empty API key is now accepted, the Authorization header is omitted unless configured, and a missing model name returns an error instead of being dereferenced. The current head is mergeable/CLEAN with no unresolved review threads or failed/pending checks. Focused and package-level Go tests pass, and CodeRabbit's two test findings were addressed with regression coverage.

**Review by Haruko386 — APPROVED** [review]
LGTM
